android/src/main/java/com/adyenreactnativesdk/component/BaseModuleException.kt

```diff
@@ -0,0 +1,44 @@
+package com.adyenreactnativesdk.component
+
+open class KnownException @JvmOverloads constructor(
+    val code: String,
+    errorMessage: String,
+    cause: Throwable? = null
+) : RuntimeException(errorMessage, cause)
+
+class BaseModuleException(code: String, message: String, cause: Throwable? = null) :
```

Comment:
you can make this base class a sealed class and make sub types extend from it. would look like this:
```
open class KnownException @JvmOverloads constructor(
    val code: String,
    errorMessage: String,
    cause: Throwable? = null
) : RuntimeException(errorMessage, cause)

sealed class BaseModuleException(code: String, message: String, cause: Throwable? = null) :
    KnownException(code = code, errorMessage = message, cause) {
    class Canceled() : BaseModuleException(code = "canceledByShopper", message = "Payment canceled by shopper")

    // Rest of the exceptions

    class NoPaymentMethod(type: String) : BaseModuleException(
        code = "noPaymentMethod",
        message = "Can not find payment method of type $type in provided list"
    )
}
```

Reply:
:+1:

Reply:
Fixed in [b32d205](https://github.com/Adyen/adyen-react-native/pull/52/commits/b32d205b75611e56ce710cb8fdd41f980d4cb88a)